doc/man3/DEFINE_STACK_OF.pod

```diff
@@ -90,18 +92,28 @@ B<NULL>.
 sk_TYPE_value() returns element B<idx> in B<sk>, where B<idx> starts at
 zero. If B<idx> is out of range then B<NULL> is returned.
 
-sk_TYPE_new() allocates a new empty stack using comparison function B<compar>.
-If B<compar> is B<NULL> then no comparison function is used.
+sk_TYPE_new() allocates a new empty stack using comparison function B<compare>.
+If B<compare> is B<NULL> then no comparison function is used. This function is
+equivalent to sk_TYPE_new_reserve(compare, 0).
 
-sk_TYPE_new_null() allocates a new empty stack with no comparison function.
+sk_TYPE_new_null() allocates a new empty stack with no comparison function. This
+function is equivalent to sk_TYPE_new_reserve(NULL, 0).
 
 sk_TYPE_reserve() allocates additional memory in the B<sk> structure
 such that the next B<n> calls to sk_TYPE_insert(), sk_TYPE_push()
 or sk_TYPE_unshift() will not fail or cause memory to be allocated
 or reallocated. If B<n> is zero, any excess space allocated in the
 B<sk> structure is freed. On error B<sk> is unchanged.
```

Comment:
On related side note. Sentence before last one seems misleading. Excess space is apparently freed whenever current data + reservation is smaller than st.num_alloc. In other words it's not specific to case of n being zero. Ping @paulidale.

Reply:
Yes, the excess space is freed in this circumstance too.  However, the guarantee that the next n additions won't cause memory allocation is true.

I don't see a need to document this _internal_ behaviour -- the stack will still function as expected.  If this were documented, it would unnecessarily restrict any future re-implementation of the stack.

Reply:
In other words the sentence in question should be simply omitted, right?

Reply:
I think so. PR #4582 is created to remove that sentence, to avoid 're-approval' of this PR...

Reply:
I'm not sure removal is required.  Being able to free up excess space seems like a reasonable thing to ask for, even without knowing anything about the internals of the stack data structure.

Reply:
Then the `Note` section might be a better place to put this. In fact, I feel it would be better either to add a new standalone API to handle the 'free' job of the excess spaces to avoid misleading, e.g. `OPENSSL_sk_shrink`...

(First time I saw `OPENSSL_sk_reserve` and I thought if I passed a `0` of `n`, it would do nothing, just as what `malloc`-like functions do. But it's not...)